app/src/main/java/com/github/swent/echo/compose/components/TopAppBar.kt

```diff
@@ -21,15 +22,23 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextOverflow
 import com.github.swent.echo.R
+import com.github.swent.echo.viewmodels.HomeScreenViewModel
+import com.github.swent.echo.viewmodels.MapOrListMode
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun TopAppBar(scope: CoroutineScope, drawerState: DrawerState, mode: MutableState<MapOrListMode>) {
+fun TopAppBar(
+    scope: CoroutineScope,
+    drawerState: DrawerState,
+    homeScreenViewModel: HomeScreenViewModel
```

Comment:
I think here the `mode` could be just of type `MapOrListMode` and not `MutableState<MapOrListMode>` makes the composable cleaner (stateless).

Reply:
True